analyzer/src/main/kotlin/MavenSupport.kt

```diff
@@ -154,12 +152,8 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
         val session = LegacyLocalRepositoryManager.overlay(localRepository, repositorySystemSession,
                 aetherRepositorySystem)
 
-        val localRepositoryManager = localRepositoryManagerConverter(session.localRepositoryManager)
-
-        return if (localRepositoryManager == session.localRepositoryManager) {
-            session
-        } else {
-            DefaultRepositorySystemSession(session).setLocalRepositoryManager(localRepositoryManager)
+        return DefaultRepositorySystemSession(session).apply {
+            setWorkspaceReader(workspaceReader)
```

Comment:
`RepositorySystemSession` has a fluent API, you don't need `apply`:
`return DefaultRepositorySystemSession(session).setWorkspaceReader(workspaceReader)`

Reply:
Done.